index.js

```diff
@@ -86,26 +86,43 @@ class MultiCoreIndexer extends TypedEmitter {
   }
 
   /**
-   * Add a core to be indexed
+   * Add a hypercore to the indexer. Must have the same value encoding as other
+   * hypercores already in the indexer.
+   *
+   * Rejects if called after the indexer is closed.
+   *
    * @param {import('hypercore')<T, any>} core
    */
   addCore(core) {
+    this.#assertOpen('Cannot add core after closing')
     const coreIndexStream = new CoreIndexStream(core, this.#createStorage)
     this.#indexStream.addStream(coreIndexStream)
   }
 
   /**
-   * Resolves when indexing state is 'idle'
+   * Resolves when indexing state is 'idle'.
+   *
+   * Resolves if the indexer is closed before this resolves. Rejects if called
+   * after the indexer is closed.
```

Comment:
I wasn't sure whether `idle()` should resolve or reject in the following case:

```js
const idlePromise = indexer.idle()
indexer.close()
await idlePromise
```

I decided to make it _resolve_, but I don't feel strongly.

Reply:
in this specific case, i'd say resolving makes sense